client/src/modules/dashboard/shared/components/Layout.tsx

```diff
@@ -17,17 +22,23 @@ export const Layout = ({ children }: { children: React.ReactNode }) => {
   return (
     <>
       <HStack as="nav" my="2">
-        {links.map((item) => (
-          <LinkButton
-            key={item.link}
-            href={item.link}
-            colorScheme={
-              router.pathname.startsWith(item.link) ? 'blue' : undefined
-            }
-          >
-            {item.text}
-          </LinkButton>
-        ))}
+        {links
+          .filter(
+            ({ permission }) =>
+              typeof permission === 'undefined' ||
+              useCheckPermission(permission),
```

Comment:
```suggestion
            ({ requiredPermission }) =>
              !requiredPermission || useCheckPermission(requiredPermission),
```
I think this reads a little better and (particularly if we use `Permission`) we should not need to worry about the distinction between `''` and `undefined`.